api/server/server.go

```diff
@@ -193,16 +194,34 @@ func postContainersKill(eng *engine.Engine, version version.Version, w http.Resp
 	if vars == nil {
 		return fmt.Errorf("Missing parameter")
 	}
-	if err := parseForm(r); err != nil {
+	err := parseForm(r)
+	if err != nil {
 		return err
 	}
-	job := eng.Job("kill", vars["name"])
-	if sig := r.Form.Get("signal"); sig != "" {
-		job.Args = append(job.Args, sig)
+
+	var sig uint64
+	name := vars["name"]
+
+	// If we have a signal, look at it. Otherwise, do nothing
+	if sigStr := vars["signal"]; sigStr != "" {
```

Comment:
This might be a bigger discussion (bordering on bikeshedding :-) ) but to me the api/server code is all about the http transport and I expect we'll eventually have others, so when that day comes I would want logic that isn't specific to the transport to be done within the daemon itself.  And I consider logging of a container being deleted to be the job of the deamon (or the container itself) but not the api/http/transport layer.  So, in that respect I would prefer if we didn't remove those other daemon funcs, if nothing else to have consistency and proper separation of concern.